app/src/main/java/com/example/calculatoropen/MainActivity.kt

```diff
@@ -0,0 +1,171 @@
+    }
+
+    private fun onDecimalPoint() {
+        if (lastNumeric && !lastBot) {
+            tvInput.append(".")
+            lastNumeric = false
+            lastBot = true
+        }
+    }
+
+    private fun onClear() {
+        tvInput.text = ""
+        lastNumeric = false
+        lastBot = false
+    }
+
+
+    private fun onEqual() {
+        if (lastNumeric) {
+            var tvValue = tvInput.text.toString()
+            var prefix = ""
+
+            try {
+                if (tvValue.startsWith("-")) {
+                    prefix = "-"
+                    tvValue = tvValue.substring(1)
+                }
+
+                when {
+                    tvValue.contains("-") -> {
+                        val splitValue = tvValue.split("-")
+                        var one = splitValue[0]
+                        val two = splitValue[1]
+                        if (prefix.isNotEmpty()) {
+                            one = prefix + one
+                        }
+                        tvInput.text =
+                                removeZeroAfterDot((one.toDouble() - two.toDouble()).toString())
+                    }
+                    tvValue.contains("+") -> {
+                        val splitValue = tvValue.split("+")
+                        var one = splitValue[0]
+                        val two = splitValue[1]
+                        if (prefix.isNotEmpty()) {
+                            one = prefix + one
+                        }
+                        tvInput.text =
+                                removeZeroAfterDot((one.toDouble() + two.toDouble()).toString())
+                    }
+                    tvValue.contains("*") -> {
+                        val splitValue = tvValue.split("*")
+                        var one = splitValue[0]
+                        val two = splitValue[1]
+                        if (prefix.isNotEmpty()) {
+                            one = prefix + one
+                        }
+                        tvInput.text =
+                                removeZeroAfterDot((one.toDouble() * two.toDouble()).toString())
+                    }
+                    tvValue.contains("/") -> {
+                        val splitValue = tvValue.split("/")
+                        var one = splitValue[0]
+                        val two = splitValue[1]
+                        if (prefix.isNotEmpty()) {
+                            one = prefix + one
+                        }
+                        tvInput.text =
+                                removeZeroAfterDot((one.toDouble() / two.toDouble()).toString())
+                    }
+                }
+            } catch (e: ArithmeticException) {
+                e.printStackTrace()
+            }
+        }
+    }
+
+    private fun removeZeroAfterDot(result: String): String {
+        var value = result
+        if (result.endsWith(".0"))
+            value = result.substring(0, result.length - 2)
+        return value
+    }
```

Comment:
Viết theo cách của kotlin nhé 
```suggestion
    private fun removeZeroAfterDot(result: String): String =
        if (result.endsWith(".0")) {
            result.substring(0, result.length - 2)
        } else {
            result
       }
```